cmake/modules/SIMDExt.cmake

```diff
@@ -77,6 +77,10 @@ elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "i386|i686|amd64|x86_64|AMD64")
       if(HAVE_INTEL_SSE4_2)
         set(SIMD_COMPILE_FLAGS "${SIMD_COMPILE_FLAGS} -msse4.2")
       endif()
+      CHECK_C_COMPILER_FLAG(-maxv2 HAVE_INTEL_AVX2)
+      if(HAVE_INTEL_AVX2)
+        set(AVX2_COMPILE_FLAGS "-mavx2")
```

Comment:
how would `AVX2_COMPILE_FLAGS` be used?

Reply:
"AVX2_COMPILE_FLAGS" should have similar usage as "SIMD_COMPILE_FLAGS". When building some libraries in future, we could build it with "-mavx2"(AVX2_COMPILE_FLAGS) options. At runtime, there'll be some other mechanisms to check whether the hardware support AVX2 or not.
Currently, "SIMD_COMPILE_FLAGS" includes too many options, such as "-msse2" & "-mssse3" and so on. If we add "-mavx2" into SIMD_COMPILE_FLAGS, there'll be some unknown stability problems in jerasure.

Reply:
as discussed, i suggest use MFV (https://lwn.net/Articles/691932/) to implement this.

Reply:
I'll check it later